src/kubelet/kubelet_test.go

```diff
@@ -24,18 +28,100 @@ import (
 )
 
 func TestScraper(t *testing.T) {
+	// Network metrics are known to be missing on some environments.
+	networkMetrics := []string{"net.rxBytesPerSecond", "net.txBytesPerSecond", "net.errorsPerSecond"}
+	// TODO
+	nodeUtilizationMetrics := []string{"allocatableCpuCoresUtilization", "allocatableMemoryUtilization"}
 
-	commonMetricsToExclude := []string{"net.rxBytesPerSecond", "net.txBytesPerSecond", "net.errorsPerSecond"}
-	nodeMetricsToExclude := append(commonMetricsToExclude, "allocatableCpuCoresUtilization", "allocatableMemoryUtilization")
+	// Pods and containers that are not in a running state will not have these metrics.
+	notRunningMetrics := []string{"memoryUsedBytes", "memoryWorkingSetBytes", "cpuUsedCores",
+		"fsAvailableBytes", "fsCapacityBytes", "fsUsedBytes", "fsUsedPercent", "fsInodesFree", "fsInodes",
+		"fsInodesUsed", "containerID", "containerImageID", "isReady", "podIP"}
+
+	// Utilization metrics will not be present if the corresponding limit/request is not present.
+	utilizationDependencies := map[string][]string{
+		"cpuLimitCores":        {"cpuCoresUtilization"},
+		"cpuRequestedCores":    {"requestedCpuCoresUtilization"},
+		"memoryLimitBytes":     {"memoryUtilization"},
+		"memoryRequestedBytes": {"requestedMemoryUtilization"},
+	}
+
+	limitsRequestsRegex := regexp.MustCompile("(Limit|Requested)(Cores|Bytes)$")
 
 	// Create an asserter with the settings that are shared for all test scenarios.
-	asserter := testutil.NewAsserter().
+	asserter := asserter.New().
 		Using(metric.KubeletSpecs).
 		Excluding(
-			testutil.ExcludeMetrics("pod", commonMetricsToExclude...),
-			testutil.ExcludeMetrics("node", nodeMetricsToExclude...),
-			testutil.ExcludeOptional(),
-		).Excluding(ExcludeMissingMetricsPendingPod)
+			// Network metrics
+			exclude.Metrics(networkMetrics...),
+
+			// Node utilization  metrics.
+			exclude.Exclude(exclude.Groups("node"), exclude.Metrics(nodeUtilizationMetrics...)),
```

Comment:
I would add an helper function as getKubeletAsserter, just to divide logically some code and keep the test readable

Reply:
Addressed in https://github.com/newrelic/nri-kubernetes/pull/303/commits/dfdd7d9b14c380399cf3fe3106814b62236ba23c